packages/runtime/test/memory.test.ts

```diff
@@ -0,0 +1,323 @@
+/**
+ * IGNORED BY AVA
+ * RUN WITH pnpm test:memory
+ * */
+
+import test from 'ava';
+
+import {
+  ExecutionPlan,
+  NOTIFY_JOB_COMPLETE,
+  NotifyJobCompletePayload,
+} from '../src';
+import callRuntime from '../src/runtime';
+
+/**
+ * This file contains various memory tests for runtime usage
+ * The aim right now sis to understand how accurate process.memoryUsage()
+ * reports are at the end of the job
+ *
+ * Something to bear in mind here is that all the tests run in the same process and share memory
+ * I fully expect test 1 to have an impact on test 2
+ * Part of the testing is to understand how much
+ *
+ * Also a consideration for the engine: after a job completes, I don't think we do any cleanup
+ * of the SourceTextModule and context etc that we created for it.
+ *
+ * In the CLI this doesn't mattter because the process ends, and actually in the worker right now
+ * We burn the thread so it still doesn't matter much/
+ */
+
+test.afterEach(() => {
+  // Force gc to try and better isolate tests
+  // THis may not work and maybe we need to use threads or something to ensure a pristine environment
+  // Certainly runs seem to affect each other in the same process (no suprise there)
+  // @ts-ignore
+  global.gc();
+});
+
+type Mem = {
+  job: number; // heapUsed in bytes
+  system: number; // rss in bytes
+};
+
+// This helper will run a workflow and return
+// memory usage per run
+const run = async (t, workflow: ExecutionPlan) => {
+  const mem: Record<string, Mem> = {};
+
+  const notify = (evt: string, payload: NotifyJobCompletePayload) => {
+    if (evt === NOTIFY_JOB_COMPLETE) {
+      mem[payload.jobId] = payload.mem;
+      logUsage(t, payload.mem, `job ${payload.jobId}`);
+    }
+  };
+
+  const state = await callRuntime(
+    workflow,
+    {},
+    {
+      strict: false,
+      callbacks: { notify },
+      globals: {
+        process: {
+          memoryUsage: () => process.memoryUsage(),
+        },
+      },
+    }
+  );
+  return { state, mem };
+};
+
+const logUsage = (t: any, mem: Mem, label = '') => {
+  // What kind of rounding should I Do?
+  // Rounding to an integer is best for humans but I think in these tests we lose a lot of fidelity
+  // I mean you could lose nearly 500kb of accuracy, that's a lot!
```

Comment:
I think MB to 2 decimal places is probably fine hey? You lose at most 49KB (if my understanding is correct).

Reply:
Yep, thanks! I'll tidy this comment up - it's a bit too stream-of-consciousness...